src/sagemaker_sklearn_extension/preprocessing/encoders.py

```diff
@@ -620,3 +621,248 @@ def inverse_transform(self, X):
                 X_tr[unknown_mask, idx] = None
 
         return X_tr
+
+
+class WOEAsserts(Enum):
+    ALPHA = "Regularization parameter `alpha` must be non-negative."
+    BINARY = "Weight-of-Evidence encoder is only supported for binary targets."
+    BINNING = "Binning strategy must be in {'uniform', 'quantile', 'kmeans'}."
+    NBINS = "Number of bins must be larger than 2."
+    MISSING = "Weight-of-Evidence encoder does not support missing values."
+    DIMS_TRANSFORM = "The number of columns of `X` doesn't match the encoder."
+    FEAT_OOR = "At least one of the feature index is out of the allowed range."
+
+
+class WOEEncoder(BaseEstimator, TransformerMixin):
+    """Weight of Evidence (WoE) encoder: encodes categorical features as a
+    numerical vector using weight of evidence encoding. This is only supported
+    with binary targets.
+    Both the features and the target are assumed to be free of missing values,
+    missing values should be handled separately before the encoding.
+    A binning function can be provided to handle numerical features which are
+    then binned first then encoded.
+    Note that the sign of the weight of evidence values depends on the
+    order in which the categories of the target column are detected. This does
+    not affect the performance of any supervised model applied thereafter.
+    See [1] for more details on WoE.
+
+    Parameters
+    ----------
+    feature_indices: list of integer indices (default=None)
+        Indices of the features to encode with WoE. If set to `None`, all
+        features will be encoded.
+
+    binning: {'uniform', 'quantile', 'kmeans', None}, default=None
+        What binning method to apply if not set to None. See [2].
+
+    n_bins: int (default=10), greater than 2
+        Number of bins to use if binning is to be applied.
+
+    alpha: float (default = 0.5), non-negative
+        Regularization value to avoid numerical errors due to division by
+        zero in the computation of the weight of evidence (e.g. in case the
+        data points corresponding to one category of a feature all have
+        the same target value).
+
+    laplace: boolean (default = False)
+        If alpha is positive, adds Laplace smoothing to the computation
+        of the weight of evidence.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> from woe import WOEEncoder
+    >>> sex = np.random.choice(['m', 'f'], size=100)
+    >>> age = np.random.randint(low=25, high=95, size=100)
+    >>> y = np.random.choice([0, 1], size=100)
+    >>> e_sex = WOEEncoder().fit_transform(sex, y)
+    >>> e_age = WOEEncoder(binning='quantile', n_bins=5).fit_transform(age, y)
+
+    Attributes
+    ----------
+    _beta: float
```

Comment:
Documentation shows `_beta` but code shows `beta_`. Documentation needs to be updated